Cleanups for Lasagna exercise and basics concept from feedback. (#1766)
exercism/go: exercises/concept/lasagna/.docs/hints.md

- An [integer value][integers] can be defined as one or more consecutive digits.

> If you see a `panic:` error when running the tests, this is as you haven't implemented one of the functions (it should say which one) or you have left the boilerplate in place, you need to remove the `panic(...)` line from the supplied code.

Reviewer: ```suggestion
If you see a `panic:` error when running the tests, this is because you have not implemented one of the functions (it should say which one) or you have left the boilerplate in place.
You need to remove the `panic(...)` line from the supplied code and replace it with a real implementation.
```

## 1. Define the expected oven time in minutes

- You need to define a [constant][constants] and assign it the expected oven time in minutes.

> If you see an `undefined: OvenTime` error then double check that you have the constant defined.

Reviewer 2: I wonder if these should be hints?

Author: Good point, the spec says these should be bullet points!

> If you see an `invalid operation: got != tt.expected (mismatched types float64 and int)` error then you've put a decimal point into the `OvenTime` - remove the decimal so that it'll be an integer.

Reviewer: ```suggestion
If you see an `invalid operation: got != tt.expected (mismatched types float64 and int)` error then you have likely put a decimal point into the `OvenTime` causing Go to infer the type as a floating point number.
Remove the decimal and the type will be inferred as an `int`.
```

> If you see a `syntax error: non-declaration statement outside function body` error then this is as you forgot the `const` keyword.

Reviewer: ```suggestion
If you see a `syntax error: non-declaration statement outside function body` error then it is likely that you forgot the `const` keyword.
```

> If you see an `syntax error: unexpected :=, expecting =` error then this is as you are trying to assign to the constant like a variable; contants are assigned using `=` not `:=`.

Reviewer: ```suggestion
If you see a `syntax error: unexpected :=, expecting =` error then you are likely trying to assign the constant using `:=` like a variable; constants are assigned using `=` not `:=`.
```
